### PR TITLE
KF 1.3 on Openshift

### PR DESCRIPTION
**Pull request by nakfour**
This PR is to get KF 1.3 working on Openshift using Open Data Hub operator.

### CODE REVIEW

**Comment by moficodes** [comment]
@nakfour ran it on openshift 4.5 on IBM Cloud. Have you tested with KFP Tekton? 

Also any plans on the multiuser version of this?

**Comment by tks-ai** [comment]
@nakfour 
im also curious about the KFP Tekton,i remember that KFP Tekton will be updated when kubeflow 1.3 is released(written on OpenDataHub Description if im correct).
but after a quick look , KFP Tekton seems not be included at the moment (commented out from kfctl_openshift_v1.3.0.yaml).
i want to know the further step of KFP Tekton on kubeflow Openshift  in order to make a decision whether to drop KFP Argo which seems not quite compatible with OCP for now. 
looking forward to your reply ,thanks.

**Comment by nakfour** [comment]
@moficodes @tks-ai is KFP-Tekton is ready I can add it in. @moficodes is is ready? If yes where is it? We can add mulit-user pipeline later, we did not have it before.

**Comment by nakfour** [comment]
@moficodes there is an issue with KFP-Tekton, opened it here https://github.com/kubeflow/manifests/issues/1856
If this takes time to fix, I am going to have to leave it out of this release

File: distributions/stacks/openshift/application/kfp-tekton/kustomization.yaml [hunk 1]
```diff
@@ -2,23 +2,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 namespace: kubeflow
```
Comment (Tomcli) [inline distributions/stacks/openshift/application/kfp-tekton/kustomization.yaml:3]:
```suggestion
```
The Tekton deployment from kfp-tekton is designed to deploy in the `tekton-pipelines` namespace, so we don't want to override it in this kustomization.

Most of the Tekton documentation is assuming it deploys in `tekton-pipelines`. And many Tekton community extensions such as tektoncd/experimental and tektoncd/catalog have this assumption as well. Also the Tekton ko and releases are based on `tekton-pipelines` namespace which helps us easier to update Tekton.

**Comment by Tomcli** [comment]
also for https://github.com/kubeflow/manifests/blob/master/distributions/stacks/openshift/application/openshift/openshift-scc/tekton-anyuid-scc.yaml

We need to include `- system:serviceaccount:$(TEKTON-NAMESPACE):tekton-pipelines-webhook` for the new release according to the Tekton documentation
https://github.com/tektoncd/pipeline/blob/main/docs/install.md#installing-tekton-pipelines-on-openshift

**Comment by nakfour** [comment]
Thanks @Tomcli now it's working.

**Comment by animeshsingh** [comment]
thanks @nakfour and @Tomcli  

@nakfour officially multi-user is not something you are looking at for this release? 

@moficodes do we have the issue open to discuss how profile controller can be extended to support multi-user on OpenShift?

**Comment by moficodes** [comment]
@animeshsingh we do not have an issue at the moment. I will open one.

**Comment by moficodes** [comment]
#1858

**Comment by nakfour** [comment]
@animeshsingh yes multi-user is not an option for this release, but its on our list to get it working. I am bumping into a couple of issues, like this one: https://github.com/kubeflow/kubeflow/issues/5271 I need more time to get it working.

**Comment by moficodes** [comment]
@nakfour They way the kfdef is laid out now, its not possible to deploy this using kustomize and component by component. 

It would be nice if we reorganized and people had a choice of using kustomize and deploy component by component.

**Comment by moficodes** [comment]
Somehow the second menu for pipeline is back. is it something you configured? @nakfour 
screenshot

**Comment by nakfour** [comment]
@moficodes no, we didn't do anything for the dashboard

**Comment by nakfour** [comment]
@moficodes we deploy from the operators, users just add delete components from the kfdef.

**Comment by moficodes** [comment]
@nakfour thats weird. In your deployment do you also see this? 

In the new release the pipeline navigation was removed and added at the top level. Somehow I am finding this ui is back for openshift. On IKS I do not see this.

**Comment by nakfour** [comment]
Where was that change done? Is it ml-pipeline-ui or central dashboard. We did not change any of that, but we did use single user pipelines.

**Comment by Tomcli** [comment]
> Where was that change done? Is it ml-pipeline-ui or central dashboard. We did not change any of that, but we did use single user pipelines.

For both KFP and KFP Tekton single user, the KFP side menu will be there by default. To remove this, patch this env variable to the ml-pipeline-ui
```
        - name: DEPLOYMENT
          value: KUBEFLOW
```
See multi-user setup for how to patch this
https://github.com/kubeflow/manifests/blob/master/apps/kfp-tekton/upstream/base/installs/multi-user/pipelines-ui/deployment-patch.yaml#L17-L18

**Comment by LaVLaS** [comment]
/lgtm

This deployment worked for me. I tested the 
* `notebook-controller` with a custom OpenShift notebook `quay.io/kubeflow/tf-notebook-image:latest` with no issues. The nb Volumes where listed under the `Volumes` and could be deleted successfully
* The default KFP Argo pipelines worked with the `demo flip-coin` pipeline

**Comment by yhwang** [comment]
one issue on the `notebooks`
when I tried to connect to the notebook I created, it showed me this error in a new tab:
```
upstream connect error or disconnect/reset before headers. reset reason: connection failure
```

**Comment by nakfour** [comment]
@yhwang we have specific notebooks built just for OCP, please use: quay.io:kubeflow/tf-notebook-image:latest

**Comment by yhwang** [comment]
@nakfour yes, the custom image works. but it's old jupyter notebook. hope it would be updated to jupyterlab soon. Thanks!

**Comment by nakfour** [comment]
Thanks @Tomcli added the change to get rid of double UI menu

**Comment by nakfour** [comment]
@yhwang yes we will update it, next on our list, opened an issue here for it: https://github.com/opendatahub-io/manifests/issues/68

**Review by crobby — APPROVED** [review]
This appears to be working as advertised.

**Pull request merged**